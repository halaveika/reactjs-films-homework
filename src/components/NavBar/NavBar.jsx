import React from 'react';
import { Row, Button } from 'antd';
import './NavBar.scss';

export default function NavBar() {
  return (
    <Row className="navbar-container" justify="space-between">
      <Button>Watch Now</Button>
      <Button>View Info</Button>
    </Row>
  );
}
