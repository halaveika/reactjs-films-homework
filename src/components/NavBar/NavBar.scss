.navbar-container {
  width: 220px;

  button {
    background-color: transparent;
    color: #fff;
  }
}
