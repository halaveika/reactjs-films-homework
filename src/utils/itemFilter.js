const itemfilter = (item, genre) => item
  .filter((e) => ((genre) ? e.genres.filter((name) => name === genre).length > 0 : e));

export default itemfilter;
