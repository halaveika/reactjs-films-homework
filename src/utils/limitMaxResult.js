import { MAX_SEARCH_RESULT, MAX_PAGE } from '../constants';

const limitMaxResult = (result) => {
  if (result.total_pages > MAX_PAGE) { result.total_pages = MAX_PAGE; }
  if (result.total_results > MAX_SEARCH_RESULT) { result.total_results = MAX_SEARCH_RESULT; }
  return result;
};

export default limitMaxResult;
