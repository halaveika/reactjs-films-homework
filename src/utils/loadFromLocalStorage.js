const loadFromLocalStorage = () => {
  const serializedState = localStorage.getItem('state');
  if (serializedState === null) return undefined;
  return JSON.parse(serializedState);
};

export default loadFromLocalStorage;
