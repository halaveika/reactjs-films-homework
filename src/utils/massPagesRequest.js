import {API_PAGE_SIZE} from '../constants';

const massPagesRequest = async(page, pageSize, httpService, str='') => {
  console.log('page: ' + page);
  console.log('pageSize: ' + pageSize);
  console.log('str: ' + str);
  const countPages = Math.ceil(pageSize / API_PAGE_SIZE);
  console.log('countPages: ' + countPages);
  let json;
  if(str){
    json = await Promise.all(Array(countPages).fill(1).map(async(_, index) =>  await httpService(str, page + index)));
  } else {
    json = await Promise.all(Array(countPages).fill(1).map(async(_, index) =>  await httpService(page + index)));
  }
  return {
    page: page,
    results: json.reduce((arr,e) => arr = arr.concat(e.results),[]),
    total_pages: json[json.length - 1].total_pages,
    total_results: json[json.length - 1].total_results,
  }

}

export default massPagesRequest;
