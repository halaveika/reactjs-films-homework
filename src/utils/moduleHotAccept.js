module.exports = {
  moduleHotAccept(mod) {
    if (mod && mod.hot) {
      mod.hot.accept();
    }
  },
};
