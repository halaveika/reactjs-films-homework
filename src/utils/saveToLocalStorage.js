const saveToLocalStorage = (state) =>{
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  }

  export default saveToLocalStorage
