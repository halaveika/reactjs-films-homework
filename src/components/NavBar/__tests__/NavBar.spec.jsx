import React from 'react';
import { create, act } from 'react-test-renderer';
import '../../../utils/matchMedia';
import NavBar from '..';

const mockData = {
  overview: 'There are growing dangers in the wizarding world of 1926 New York. Something mysterious is leaving a path of destruction in the streets, threatening to expose the wizarding',
};

describe('test NavBar component', () => {
  it('should render NavBar component', () => {
    const component = create(
      <NavBar
        overview={mockData.overview}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handleVisibleChange should change state', () => {
    const { root } = create(
      <NavBar
        overview={mockData.overview}
      />,
    );
    expect(root.instance.state.visible).toBe(false);
    act(() => { root.findAllByType('button')[1].props.onClick(); });
    expect(root.instance.state.visible).toBe(true);
  });
});
