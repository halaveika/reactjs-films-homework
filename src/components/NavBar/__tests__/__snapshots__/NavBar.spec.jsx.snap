// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test NavBar component should render NavBar component 1`] = `
<div
  className="ant-row ant-row-space-between navbar-container"
  style={Object {}}
>
  <button
    className="ant-btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      Watch Now
    </span>
  </button>
  <button
    className="ant-btn"
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span>
      View Info
    </span>
  </button>
</div>
`;
